### ISSUE TITLE
Unnecessary `&mut` should be warned

### ISSUE THREAD

**Issue by upsuper** [rust-lang/rust#53759]
Given the following code:
```rust
fn foo(_: &mut usize) {}
let mut a = 0;
let mut b = &mut a;
foo(&mut b);
```
there is no warning saying that the `&mut` in the last line is unnecessary.

Some novice don't always have clear idea about where should have `&mut` so they may add lots of unnecessary ones which leads to more unnecessary `mut` variables.

It might be helpful if we can warn on this case so that they know this isn't necessary.

**Comment by csmoe** [comment]
I doubt this since the last `&mut b` is trying to pass a mutable reference of a mutable reference, sometimes this may be an intended behavior.

**Comment by ivanbakel** [comment]
This behaviour also happens in the simpler case

    fn foo (_: &i32) {}
    let mut a = 0;
    foo (&mut a);

Which looks a lot worthier of a lint.

**Comment by sanxiyn** [comment]
This is rust-lang/rust-clippy#1230. I agree with @ExpHP that this belongs in clippy and not rustc.
